icu4j/main/tests/core/src/com/ibm/icu/dev/test/format/PluralRulesTest.java

```diff
@@ -929,6 +930,45 @@ public void TestKeywords() {
         }
     }
 
+
+
+    @Test
+    public void testCompactDecimalPluralKeyword() {
+        PluralRules rules = PluralRules.createRules("one: i = 0,1 @integer 0, 1 @decimal 0.0~1.5;  many: e = 0 and i % 1000000 = 0 and v = 0 or " +
+                "e != 0 .. 5;  other:  @integer 2~17, 100, 1000, 10000, 100000, 1000000, @decimal 2.0~3.5, 10.0, 100.0, 1000.0, 10000.0, 100000.0, 1000000.0, …");
+        ULocale locale = new ULocale("fr-FR");
+
+        // simple notation (no rounding from compact notation)
+        Notation simple = Notation.simple();
+        assertEquals("PluralRules select, no suppressed exponent", "one", getPluralKeyword(rules, locale, 0, simple));
+        assertEquals("PluralRules select, no suppressed exponent", "one", getPluralKeyword(rules, locale, 1, simple));
```

Comment:
I personally prefer the style of parameterized tests.  We have JUnitParams available if you want to use that.  Otherwise, I find tests written as follows to be more readable and maintainable:

```java
Object[][] cases = new Object[][] {
    { "one", 0 },
    { "other", 2 },
    // ...
};
for (Object[] cas : cases) {
    String expected = (String) cas[0];
    int input = (Integer) cas[1];
    String message = "PluralRules select, no suppressed exponent: " + input;
    String actual = getPluralKeyword(rules, locale, input, simple);
    assertEquals(message, expected, actual);
}
```

It's important to make the message contain the parameters since the line number isn't as useful when the test is parameterized.

Reply:
Done.